File: ck_kjwj.py
# -*- coding: utf8 -*-
"""
cron: 17 7 * * *
new Env('科技玩家');
"""
import requests
import os
import time

from notify_mtr import send
from utils import get_data

info= ""

class KJWJ:
    def login(usr, pwd):
        session = requests.Session()
        login_url = 'https://www.kejiwanjia.com/wp-json/jwt-auth/v1/token'
        headers = {
            'user-agent': 'Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.198 Safari/537.36'
        }
        data = {
            'nickname': '',
            'username': usr,
            'password': pwd,
            'code': '',
            'img_code': '',
            'invitation_code': '',
            'token': '',
            'smsToken': '',
            'luoToken': '',
            'confirmPassword': '',
            'loginType': ''
        }
        respon = session.post(login_url, headers=headers, data=data)
        if respon.status_code == 200:
            status = respon.json()
            info += f"账号：{status.get('name')}登陆成功\n"
            info += f"ID：{status.get('id')}\n"
            info += f"金币：{status.get('credit')}\n"
            info += f"等级：{status.get('lv').get('lv').get('name')}\n"
            token = status.get('token')
            check_url = 'https://www.kejiwanjia.com/wp-json/b2/v1/userMission'
            check_head = {
                'authorization': f'Bearer {token}',
                'origin': 'https://www.kejiwanjia.com',
                'referer': 'https://www.kejiwanjia.com/task',
                'user-agent': 'Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.198 Safari/537.36'
    
            }
            respons = session.post(check_url, headers=check_head)
            if respons.status_code == 200:
                desp = respons.json()
                if type(desp) == str:
                    info += f"已经签到：{info}金币\n\n"
                else:
                    info += f"签到成功：{info.get('credit')}金币\n\n"
        else:
            info += f"账号登陆失败: 账号或密码错误\n\n"
        return info
    
    def main(self):
        i = 0
        msg = ""
        msg += "检测到 " + str(len(self.check_items)) + " 个科技玩家帐户，开始签到："
        for check_item in self.check_items:
            username = check_item.get("username")
            password = check_item.get("password")
            try:
                msg += self.login(username, password)
            except Exception as e:
                msg += "程序执行异常：" + str(e)
        return msg

if __name__ == "__main__":
    data = get_data()
    _check_items = data.get("KJWJ", [])
    res = KJWJ(check_items=_check_items).main()
    send("科技玩家", res)

File: ck_kjwj.js
/*
31 7 * * * ck_kjwj.js
*/
const utils = require('./utils');
const Qs = require('qs');
const Env = utils.Env;
const getData = utils.getData;
const $ = new Env('科技玩家');
const notify = $.isNode() ? require('./notify') : '';
const AsVow = getData().KJWJ;
var info = '';
var desp = '';
var token = '';


const headers = {
    'Host': 'www.kejiwanjia.com',
    'Referer': 'https://www.kejiwanjia.com/mission/today',
    'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.198 Safari/537.36'
};

const data = {
    'username': '',
    'password': ''
};


kjwj();

async function kjwj() {
  if (AsVow) {
    for (i in AsVow) {
      username = AsVow[i].username;
      password = AsVow[i].password;
      data['username'] = username;
      data['password'] = password;
      if (username && password) {
        head = `=== 正对在 ${username} 的账号签到===\n`;
        info += `\n${head}`;
        await getauth();
        await pre_sign();
        await sign();
        desp += info;
        info = '';
      } 
    }
    info += desp;
    console.log(info);
    notify.sendNotify('科技玩家', info);
  } else {
    info = '签到失败：请先获取Cookie⚠️';
    console.log(info);
    notify.sendNotify('科技玩家', info);
  }
}

function getauth() {
  url = 'https://www.kejiwanjia.com/wp-json/jwt-auth/v1/token';
  headers['Referer'] = 'https://www.kejiwanjia.com/';
  headers['Content-Type'] = 'application/x-www-form-urlencoded';
  const request = {
      url: url,
      headers: headers,
      body: Qs.stringify(data)
  };
  return new Promise(resolve => {
    $.http.post(request)
      .then(async (resp) => {
        resdata = $.toObj(resp.body);
        info += `账号：${resdata.name}\n`;
        info += `ID：${resdata.id}\n`;
        info += `金币：${resdata.credit}\n`;
        info += `等级：${resdata.lv.lv.name}\n`;
        token = resdata.token;
      })
      .catch((err) => {
        const error = '账号信息获取失败⚠️';
        console.log(error + '\n' + err);
      })
      .finally(() => {
        resolve();
      });
  });
}

function pre_sign() {
  url = 'https://www.kejiwanjia.com/wp-json/b2/v1/getUserMission';
  headers['Content-Type'] = 'application/x-www-form-urlencoded';
  headers['Origin'] = 'https://www.kejiwanjia.com/';
  headers['Authorization'] = `Bearer ${token}`;
  suff = 'count=10&paged=1';
  const request = {
      url: url,
      headers: headers,
      body: suff
  };  
  return new Promise(resolve => {
    $.http.post(request)
      .then(async (resp) => {
        resdata = $.toObj(resp.body);
        if (resdata.mission.credit != "0") {
          info += `今天已签到：获得${resdata.mission.credit}金币\n\n`;
        }
      })
      .catch((err) => {
        const error = '🆕--签到前--状态获取失败⚠️';
        console.log(error + '\n' + err);
      })
      .finally(() => {
        resolve();
      });
  });
}

function sign() {
  url = 'https://www.kejiwanjia.com/wp-json/b2/v1/userMission';
  headers['Origin'] = 'https://www.kejiwanjia.com/';
  headers['Authorization'] = `Bearer ${token}`;
  const request = {
      url: url,
      headers: headers
  };
  return new Promise(resolve => {
    $.http.post(request)
      .then(async (resp) => {
        resdata = $.toObj(resp.body);
        if ((typeof resdata) != 'string') {
          info += `每日首次签到成功：获得${resdata.credit}金币\n\n`;
        }
      })
      .catch((err) => {
        const error = '🆕--签到--状态获取失败⚠️';
        console.log(error + '\n' + err);
      })
      .finally(() => {
        resolve();
      });
  });
}
module.exports = kjwj;
